### PR TITLE
Add support for private_key_file and private_key_file_pwd params

### LINKED ISSUE #1565: SNOW-818558: Support configuring file paths for private key authentication
## What is the current behavior?

Python Connector requires manually reading and decrypting a private key file (and optional passphrase), documented at https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-example#using-key-pair-authentication-key-pair-rotation

Passing file paths and passphrases directly into the config is not currently supported

The .NET driver supports such an option (via `PRIVATE_KEY_FILE` and `PRIVATE_KEY_PWD`): https://github.com/snowflakedb/snowflake-connector-net#create-a-connection

The ODBC driver also supports such options (via `PRIV_KEY_FILE` and `PRIV_KEY_FILE_PWD`): https://docs.snowflake.com/en/developer-guide/odbc/odbc-parameters#using-key-pair-authentication

## What is the desired behavior?

Support convenience options (e.g. `private_key_file` and `private_key_file_passphrase`) that perform reading and decrypting private key files automatically, and reduces the work required to use key-pair authentication with the connector.

## How would this improve `snowflake-connector-python`?

These options would simplify configuring private key authentication in higher level applications such as SQLAlchemy, and Django backend for Snowflake (related issues: https://github.com/Snowflake-Labs/django-snowflake/issues/67 and https://github.com/Snowflake-Labs/django-snowflake/issues/72)

### PR DESCRIPTION
**Pull request by bennylu2**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #1565

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [x] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

Adding support for specifying a private key file and password for authenticating with Snowflake. Connector already supports `private_key` but it must be provided as bytes. The JDBC driver supports these as the parameters [private_key_file](https://docs.snowflake.com/en/developer-guide/jdbc/jdbc-parameters#private-key-file) and `private_key_file_pwd` but is missing from the `snowflake-connector-python` project namely from `auth/keypair.py`

### CODE REVIEW

**Comment by bennylu2** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/snowflake/connector/auth/keypair.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
     Encoding,
     PublicFormat,
     load_der_private_key,
+    load_pem_private_key
```
Comment (bennylu2) [inline src/snowflake/connector/auth/keypair.py:22]:
```suggestion
    load_pem_private_key,
```

**Comment by sfc-gh-mkeller** [comment]
I'd prefer if `src/snowflake/connector/auth/keypair.py` was left untouched, and we introduced all the logic in the `connection.py`.
We can just read in the key and decrypt it and then treat it like it was provided to the connection as straight `bytes`.

**Comment by bennylu2** [comment]
> I'd prefer if `src/snowflake/connector/auth/keypair.py` was left untouched, and we introduced all the logic in the `connection.py`. We can just read in the key and decrypt it and then treat it like it was provided to the connection as straight `bytes`.

Refactored, is this what you were thinking?

**Comment by sfc-gh-mkeller** [comment]
Yes, that looks wonderful! Please add a changelog entry and a unit test for the loading the private key. I'd love to accept this afterwards!

**Comment by bennylu2** [comment]
> Yes, that looks wonderful! Please add a changelog entry and a unit test for the loading the private key. I'd love to accept this afterwards!

Done! I noticed there's not really tests for connection.py so I made the loading private key a function to write a unit test. Lmk if there's something I missed

**Comment by bennylu2** [comment]
Could the workflows be enabled please? Would be good to see if there's any failing workflows/checks

**Comment by sfc-gh-mkeller** [comment]
Merged through #1808 
❤️ @bennylu2

**Pull request closed without merging**